lib/logstash/outputs/kusto/custom_size_based_buffer.rb

```diff
@@ -0,0 +1,50 @@
+module LogStash
+    module Outputs
+      class CustomSizeBasedBuffer
+        def initialize(max_size, max_interval, &flush_callback)
+          @max_size = max_size
+          @max_interval = max_interval
```

Comment:
Considering the flush thread sleeps for the entire @max_interval. If the interval is large, it might introduce a considerable delay the buffer flush. Can we do something like this

def start_flusher_thread
  @flusher_thread = Thread.new do
    loop do
      sleep @max_interval
      flush_if_needed
    end
  end
end

def flush_if_needed
  @mutex.synchronize do
    if Time.now - @last_flush_time >= @max_interval
      flush
    end
  end
end